groovy-commons/src/main/groovy/yakworks/commons/map/PathKeyMap.groovy

```diff
@@ -74,7 +74,7 @@ class PathKeyMap<K,V> implements Map<K,V>, Cloneable  {
                     clonedMap[k] = (val as PathKeyMap).cloneMap()
                 }
                 // if its a list of PathKeyMaps then iterate over and clone those too
-                else if(val instanceof Collection<PathKeyMap>) {
+                else if(val instanceof Collection && isCollectionOfPathKeyMaps(val)) {
```

Comment:
see comment below. and its is a collection then how does clone happen if we skip it? does it matter since it will be a ref of does clone take place below?